client/image_pull.go

```diff
@@ -2,19 +2,46 @@ package client
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"io"
+	"iter"
 	"net/url"
 	"strings"
 
 	cerrdefs "github.com/containerd/errdefs"
 	"github.com/distribution/reference"
+	"github.com/moby/moby/client/pkg/jsonmessage"
 )
 
+type ImagePullResponse struct {
+	io.ReadCloser
+}
+
+// Messages decodes the response stream as a sequence of JSONMessages
+func (r *ImagePullResponse) JSONMessages() iter.Seq2[jsonmessage.JSONMessage, error] {
```

Comment:
Also to be clear w.r.t. my other comment; this SGTM, because we're not in the jsonmessage package here, so it's ok to be explicit.